service/collector.go

```diff
@@ -172,6 +172,11 @@ LOOP:
 	for {
 		select {
 		case err := <-col.set.ConfigProvider.Watch():
+			if !col.set.EnableConfigHotReload {
+				col.service.telemetrySettings.Logger.Warn("Config updated, but hot reloading is disabled, ignoring")
+				continue
+			}
```

Comment:
Not sure about this, it may spam the logging system.

Reply:
It might, but I think it's still useful information that config sources keep changing and the collector is ignoring them. In most cases I think this will alert users that they need to enable reloading if they want it.

Reply:
And if they don't want auto-reload? Do we just spam with "warnings"?

Reply:
Is there any situation where this is actually going to be spam? To begin with, it should be very rare for a config source to change very frequently. My thinking was that this would rarely produce an unwanted message for some users, and catch misconfigurations for others. If you think that's not a good tradeoff, I can remove it, it's not that important.

Reply:
INFO level would be more appropriate. On Kubernetes, those events can be quite frequent, as there are 3 or 4 levels of symlink between the mount path and the actual file, and any changes to that would trigger an event.

Reply:
After moving the change to Resolver, this doesn't exist anymore, as we don't get the notifications to begin with.